CONFIG_MIPS_VPE_LOADER requires to be provided with physical_memsize for VPE1

diff --git a/arch/mips/kernel/prom.c b/arch/mips/kernel/prom.c
--- a/arch/mips/kernel/prom.c
+++ b/arch/mips/kernel/prom.c
@@ -25,6 +25,10 @@
 
 static char mips_machine_name[64] = "Unknown";
 
+#ifdef CONFIG_SOC_XWAY
+unsigned long physical_memsize = 0;
+#endif
+
 __init void mips_set_machine_name(const char *name)
 {
 	if (name == NULL)
@@ -49,6 +53,10 @@ int __init early_init_dt_scan_memory_arc
 
 void __init early_init_dt_add_memory_arch(u64 base, u64 size)
 {
+#ifdef CONFIG_SOC_XWAY
+	physical_memsize = size;
+	printk ("adding memory size:%d from DT\n", (unsigned int)physical_memsize);
+#endif
 	return add_memory_region(base, size, BOOT_MEM_RAM);
 }
 
